### PR TITLE
feat: single executable distribution

### PR DESCRIPTION
**Pull request by Teddy-Schmitz**
Adds feat for: https://github.com/authcompanion/authcompanion/issues/44

Todos:
- [x] Fix integration tests
- [ ] Update documentation for new executables
- [x] Remove docker hub repo and github action for container img distribution

### CODE REVIEW

File: services/login.ts [hunk 1]
```diff
@@ -45,8 +44,8 @@ export const login = async (ctx: Context) => {
 
   //Check their password is correct, then issue access token
   if (await compare(password, user.password)) {
-    const userAccesstoken = await jwtHandler.makeAccesstoken(user);
-    const userRefreshtoken = await jwtHandler.makeRefreshtoken(user);
+    const userAccesstoken = await ctx.state.jwt.makeAccesstoken(user);
```
Comment (authcompanion) [inline services/login.ts:47]: Getting this error when logging a user in:
`ERROR TypeError: ctx.state.jwt.makeAccesstoken is not a function`

Repo: 
- deno run -A --unstable bin/cmd.ts
- then login with user on http://localhost:3002/client/v1/login

File: services/login.ts [hunk 2]
```diff
@@ -45,8 +44,8 @@ export const login = async (ctx: Context) => {
 
   //Check their password is correct, then issue access token
   if (await compare(password, user.password)) {
```
Comment (authcompanion) [inline services/login.ts:46]: workers are currently not supported in standalone binaries. we'd have to change to sync/blocking for all these password compares. So `compare` to `compareSync` per https://deno.land/x/bcrypt#syncblocking. or we find a new lib to replace this bcrypt.

reference: https://github.com/denoland/deno/issues/8654

File: bin/cmd.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  .description("Setup the authcompanion service")
+  .action(async (): Promise<void> => await setup())
+  .command("generate-key")
+  .description("Generate a new keyfile")
+  .action(async (): Promise<void> => {
+    let path = `${Deno.cwd()}/keyfile`;
+    if (config.KEYPATH != undefined) {
+      path = config.KEYPATH;
+    }
+    return await generateKey(path);
```
Comment (authcompanion) [inline bin/cmd.ts:22]: Let's log a message that the key was created successfully

File: bin/setup.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  const cwd = Deno.cwd();
+
+  const defaultDev = {
+    port: 3002,
+    log_level: "DEBUG",
+    log_handler: "console",
+    origin: "http://localhost:3001",
+    secure: false,
+    recovery_redirect_url: "http://localhost:3001/recover/",
+    db_uri: `${cwd}/authcompanion_users.db`,
```
Comment (authcompanion) [inline bin/setup.ts:15]: Let's consider making this a relative path?

Comment (Teddy-Schmitz) [reply bin/setup.ts:15]: it is a relative path, cwd = current working directory

**Pull request merged**